engine/libsuppliers.go

```diff
@@ -79,6 +79,17 @@ func (sSpls *SortedSuppliers) SortCost() {
 	})
 }
 
+// SortCost is part of sort interface,
+// sort based on Cost with fallback on Weight
+func (sSpls *SortedSuppliers) SortHighestCost() {
+	sort.Slice(sSpls.SortedSuppliers, func(i, j int) bool {
```

Comment:
this line needs recheck, not sure if we cannot integrate it in the second comparison. Maybe add a test on ==?